### PR TITLE
[AIRFLOW-5448] Handle istio-proxy for Kubernetes Pods

### PR DESCRIPTION
**Pull request by sjmiller609**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW-5448) issues and references them in the PR title.

### Description

- [x] Here are some details about my PR:

Istio service mesh is not compatible by default with Kubernetes Jobs. The normal behavior is that a Job will be started, get an istio-proxy sidecar attached to it via the istio mutating webhook, run until completion, then the 'main' container in the pod stops, but istio-proxy hangs around indefinitely. This applies to the Kubernetes Executor.

Very recently, Istio implemented an endpoint that can be called to cleanly exit the proxy, specifically designed for this use case.

explanation: https://github.com/istio/istio/issues/15041
istio PR implementing it: https://github.com/istio/istio/pull/15406
Astronomer will make a contribution to handle cleanly exit the istio-proxy by default. This will help Astronomer and other Airflow users making use of the Kubernetes Executor in combination with Istio.

Here a PR I am closing in favor of contributing to upstream: https://github.com/astronomer/airflow/pull/47

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

 - [x] All the public functions and the classes in the PR contain docstrings that explain what it does

### CODE REVIEW

**Comment by sjmiller609** [comment]
@mik-laj Thanks for your patience. This one is working for Astronomer running in Istio.

File: airflow/contrib/kubernetes/istio.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+                return False
+        # If we didn't find istio at all, then we should
+        # not shut it down. Also we should only shut it down
+        # if it has state "running".
+        return found_istio
+
+    def _shutdown_istio_proxy(self, pod):
+        """Shutdown the istio-proxy on the provided pod
+
+        Args:
```
Comment (mik-laj) [inline airflow/contrib/kubernetes/istio.py]: We use `:param:` style syntax of docstring:
https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html

File: setup.py [hunk 2]
```diff
@@ -226,6 +226,7 @@ def write_version(filename=os.path.join(*["airflow", "git_version"])):
             'thrift_sasl>=0.2.0',
             'snakebite[kerberos]>=2.7.8']
 kubernetes = ['kubernetes>=3.0.0',
+              'packaging>=19.1',
```
Comment (sjmiller609) [inline setup.py:228]: build is failing because this guy is missing on import.

Comment (sjmiller609) [reply setup.py:228]: ok it's because this is not in python2

Comment (sjmiller609) [reply setup.py:228]: i'll add the support in. one moment...

**Comment by mik-laj** [comment]
Travis is sad. Can you fix it?

**Comment by sjmiller609** [comment]
Yes I will, I have been meaning to get to it. Thank you.

**Comment by sjmiller609** [comment]
We should probably integrate into v1-10-test before -stable https://github.com/apache/airflow/pull/6171

**Pull request closed without merging**